### ISSUE TITLE
LTN tool fixes for August

### ISSUE THREAD

**Issue by dabreegster** [a-b-street/abstreet#965]
There are many TODOs for the LTN tool that should be done before end of August, but they're scattered in my notes and various old issues. This is a consolidated burndown list. CC @dingaaling 

High-priority:

- [x] An MVP way to anonymously upload a fixed copy of a proposal, recording upload date.
- [ ] Explore alternatives to browser local storage for saving; there's a 5MB limit that some users are hitting
- [x] Save changes to one-way roads and make undo/redo work there too
- [x] Some kind of tutorial / user guide. Format and contents TBD.
- [x] Updated docs on ltn.abstreet.org
- [x] Do something with the route planner UI -- it's likely to be used frequently

Low (many of these are hard not because of coding, but design/UX choices I'm not great at making):

- [x] Explain the filter types better in the UI. Maybe a big modal popup to switch between them, leaving room to explain.
- [x] Reorganize elements in the left panel and possibly split things out into several floating panels
- [x] Hideable panels, if still needed
- [x] Expose an explicit scale factor setting or other way to adjust text size. On Robin's laptop, there were no HiDPI bugs, but a real preference for smaller text
- [x] Mark roads with a bus route from OSM or GTFS. When you place a filter on one, popup a warning and ask if the user wants to make it a bus gate instead.
- [x] When trying to put a filter on a one-way, popup a warning like we do now, but offer the option to convert it to a two-way road too. Same for the freehand filter tool.
- [ ] Display one-way arrows in all views (missing in shortcuts, route planning, etc)
- [x] Consider a legend to explain map elements and colors.
- [x] Consider a "layers" tool like Ungap. Unclear about the contents.
- [ ] Improve label rendering -- performance, curving, fitting the roads, overlapping with icons & one-way arrows
- [ ] Try replacing the tooltips with a "ghost" version of what action is about to happen
- [ ] Detect illegal one-way situations
- [ ] Be able to make an entire sequence of road segments one-way -- revive the UI for selecting along a route?
- [x] Visually distinguish existing and new filters
- [x] Add zoom in/out controls as an alternate to using the mouse

**Comment by dabreegster** [comment]
"end of August"
RIP

**Comment by dabreegster** [comment]
Tasks/ideas from the session yesterday with Madison:

- [ ] Support images and have a pop-up to change filter types, which will define the term better
- [ ] Experiment with fading the shortcut color heatmap when adding filters
- [ ] Fix the top bar style -- different icon for current, remove the border
- [ ] One-way error message screen should match bus gate error screen
- [ ] Maybe explain terminology like "cell" in the legend
- [ ] Continue proposal forking prototype. Use top bar for "tabs", forget about autosave for now, maybe need a "load all" button
- [ ] Try using spotlight or another effect to highlight roads and filters that've been modified
- [ ] Tutorial mode (just point to controls, place one filter, stop east/west traffic with hints, plan some route that changes)
- [ ] Crossings + porosity (new tab, circular icons, a legend)
- [ ] Some kind of first-time splash page, ask about text size upfront

**Comment by dabreegster** [comment]
- [ ] Traffic calming / 20mph zones. This could show possible detours in the area if not all streets are calmed or filtered.
- [ ] Consider a special "filter" to mark that something isn't a likely shortcut. Service roads are an example -- based on local knowledge, sometimes they're something to worry about, sometimes not

**Comment by dabreegster** [comment]
https://a-b-street.github.io/ltn/ is the new version. All these pending tasks here are either done or already captured with another issue
